src/java/org/apache/lucene/analysis/gosen/GosenCharacterNormalizeFilter.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.lucene.analysis.gosen;
+
+import com.ibm.icu.text.Normalizer2;
+import org.apache.lucene.analysis.charfilter.BaseCharFilter;
+import org.apache.lucene.analysis.util.RollingCharBuffer;
+
+import java.io.IOException;
+import java.io.Reader;
+import java.io.StringReader;
+
+
+public class GosenCharacterNormalizeFilter extends BaseCharFilter {
```

Comment:
OK. I don't have any strong preference and opinion. To be honest, I don't think I have a good sense of naming :)
So, I will change its class name to GosenNomalizerCharFilter.